src/sentry/static/sentry/app/views/settings/organizationMembers/inviteRequestRow.tsx

```diff
@@ -0,0 +1,148 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import styled from 'react-emotion';
+
+import {Member, Organization} from 'app/types';
+import {PanelItem} from 'app/components/panels';
+import {t, tct} from 'app/locale';
+import Button from 'app/components/button';
+import Confirm from 'app/components/confirm';
+import HookOrDefault from 'app/components/hookOrDefault';
+import Tag from 'app/views/settings/components/tag';
+import Tooltip from 'app/components/tooltip';
+import space from 'app/styles/space';
+
+type Props = {
+  inviteRequest: Member;
+  inviteRequestBusy: Map<string, boolean>;
+  organization: Organization;
+  onApprove: (id: string, email: string) => Promise<void>;
+  onDeny: (id: string, email: string) => Promise<void>;
+};
+
+const InviteModalHook = HookOrDefault({
+  hookName: 'member-invite-modal:customization',
+  defaultComponent: ({onSendInvites, children}) =>
+    children({sendInvites: onSendInvites, canSend: true}),
+});
+
+type InviteModalRenderFunc = React.ComponentProps<typeof InviteModalHook>['children'];
+
+const InviteRequestRow = ({
+  inviteRequest: {id, email, inviteStatus, inviterName, roleName},
+  inviteRequestBusy,
+  organization,
+  onApprove,
+  onDeny,
+}: Props) => {
+  // eslint-disable-next-line react/prop-types
+  const hookRenderer: InviteModalRenderFunc = ({sendInvites, canSend, headerInfo}) => (
+    <StyledPanelItem>
+      <div>
+        <h5 style={{marginBottom: '3px'}}>
+          <UserName>{email}</UserName>
+        </h5>
+        {inviteStatus === 'requested_to_be_invited' ? (
+          inviterName && (
+            <Tooltip
+              title={t(
+                'An existing member has asked to invite this user to your organization'
+              )}
+            >
+              <Description>
+                {tct('Requested by [inviterName]', {inviterName})}
+              </Description>
+            </Tooltip>
+          )
+        ) : (
+          <Tooltip title={t('This user has asked to join your organization.')}>
+            <StyledTag size="small">{t('Join request')}</StyledTag>
+          </Tooltip>
+        )}
+      </div>
+      <div>{roleName}</div>
+      <ButtonGroup>
+        <Confirm
+          onConfirm={sendInvites}
+          disableConfirmButton={!canSend}
+          message={
+            <div>
+              {tct('Are you sure you want to invite [email] to your organization?', {
+                email,
+              })}
+              {headerInfo}
+            </div>
+          }
+        >
+          <Button priority="primary" size="small" busy={inviteRequestBusy.get(id)}>
```

Comment:
Damn I had no idea about the busy property. I've been using disabled a lot.. probably will need to audit that at some point